### ISSUE TITLE
paddle.fluid.core.EnforceNotMet: CUBLAS: execution failed

### ISSUE THREAD

**Issue by liushanshan07** [PaddlePaddle/Paddle#10843]
本地单机单卡训练正常。
本地单机多卡出现如下问题。
34   File "./ctc_train.py", line 125, in main
 35     train(args, data_reader=ctc_reader)
 36   File "./ctc_train.py", line 87, in train
 37     feed=get_feeder_data(data, place))
 38   File "/home/users/liushanshan/common_env/venv/lib/python2.7/site-packages/paddle/fluid/parallel_executor.py", line 238, in run
 39     @     0x7f8e11fe8e1d  google::LogMessage::Fail()
 40     self.executor.run(fetch_list, fetch_var_name)
 41 paddle.fluid.core.EnforceNotMet: CUBLAS: execution failed,  at [/home/users/shiwenguo/fluid_paddle/Paddle/paddle/fluid/operators/math/b
 42 PaddlePaddle Call Stacks:
 43 0       0x7f8e112e3486p paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int) + 486
 44 1       0x7f8e115a6532p void paddle::operators::math::Blas<paddle::platform::CUDADeviceContext>::GEMM<float>(CBLAS_TRANSPOSE, CBLAS_TRA
 45 2       0x7f8e11b5d58bp void paddle::operators::math::Blas<paddle::platform::CUDADeviceContext>::MatMul<float>(paddle::framework::Tenso
 46 3       0x7f8e11b5dd6fp paddle::operators::MulGradKernel<paddle::platform::CUDADeviceContext, float>::Compute(paddle::framework::Execut
 47 4       0x7f8e11e72ec6p paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform
 48 5       0x7f8e11da4697p
 49 6       0x7f8e11dc55e0p
 50 7       0x7f8e11dc4f75p paddle::framework::details::OpHandleBase::RunAndRecordEvent(std::function<void ()> const&) + 789
 51 8       0x7f8e11da41ffp paddle::framework::details::ComputationOpHandle::RunImpl() + 95
 52 9       0x7f8e11dc59b7p paddle::framework::details::OpHandleBase::Run(bool) + 343
 53 10      0x7f8e11dbc5eep
 54 11      0x7f8e11c73913p std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Del
 55 12      0x7f8e11c740a7p std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std:
 56 13      0x7f8e4b2ff973p pthread_once + 83
 57 14      0x7f8e11dbb502p
 58 15      0x7f8e11dbf916p std::thread::_Impl<std::_Bind_simple<ThreadPool::ThreadPool(unsigned long)::{lambda()#1} ()> >::_M_run() + 406
 59 16      0x7f8e1ff6d8a0p
 60 17      0x7f8e4b2fa1c3p
 61 18      0x7f8e4a92212dp clone + 109
 62
 63     @     0x7f8e11fec8cc  google::LogMessage::SendToLog()
 64 PC: @                0x0 (unknown)
 65 *** SIGSEGV (@0x296604) received by PID 86988 (TID 0x7f8dcce56700) from PID 2713092; stack trace: ***
 66     @     0x7f8e4b302160 (unknown)
 67     @     0x7f8e11fe8943  google::LogMessage::Flush()

**Comment by dzhwinter** [comment]
版本信息？系统环境是什么？可以贴一下复现的code吗

**Comment by liushanshan07** [comment]
cuda8 + cudnn5 + gcc-4.8.2
NCCL version：nccl_2.1.15-1+cuda8.0_x86_64
GPU: K40m Driver Version: 384.81

code:
"""Trainer for OCR CTC model."""
import os
import paddle.fluid as fluid
import dummy_reader
import ctc_reader
import argparse
from load_model import load_param
import functools
import sys
from utility import add_arguments, print_arguments, to_lodtensor, get_feeder_data
from crnn_ctc_model import ctc_train_net
import numpy as np

parser = argparse.ArgumentParser(description=__doc__)
add_arg = functools.partial(add_arguments, argparser=parser)
# yapf: disable
#add_arg('batch_size',     int,   128,     "Minibatch size.")
add_arg('batch_size',     int,  32,     "Minibatch size.")
add_arg('pass_num',       int,   100,     "# of training epochs.")
add_arg('log_period',     int,   15000,   "Log period.")
add_arg('save_model_period', int,   15000,      "Save model period. '-1' means never saving the model.")
add_arg('learning_rate',  float, 1.0e-3, "Learning rate.")
add_arg('l2',             float, 0.0004, "L2 regularizer.")
add_arg('max_clip',       float, 10.0,   "Max clip threshold.")
add_arg('min_clip',       float, -10.0,  "Min clip threshold.")
add_arg('momentum',       float, 0.9,    "Momentum.")
add_arg('rnn_hidden_size',int,   200,    "Hidden size of rnn layers.")
add_arg('device',         int,   0,      "Device id.'-1' means running on CPU"
                                         "while '0' means GPU-0.")
add_arg('model_average',     bool,   True,     "Whether to aevrage model for evaluation.")
add_arg('min_average_window',     int,   10000,     "Min average window.")
add_arg('max_average_window',     int,   15625,     "Max average window.")
add_arg('average_window',     float,   0.15,     "Average window.")
add_arg('parallel', bool, True, "Whether use parallel training.")

# yapf: disable

def load_parameter(place):
    params = load_param('./name_20conv_paddle_v1_eng.map', './results/pass-00000/')
    for name in params:
        t = fluid.global_scope().find_var(name).get_tensor()
        t.set(params[name], place)


def train(args, data_reader=dummy_reader):
    """OCR CTC training"""
    num_classes = data_reader.num_classes()
    data_shape = data_reader.data_shape()
    # define network
    images = fluid.layers.data(name='pixel', shape=data_shape, dtype='float32')
    label = fluid.layers.data(name='label', shape=[1], dtype='int32', lod_level=1)
    sum_cost, error_evaluator, model_average, decoded_out = ctc_train_net(images, label, args, num_classes)
    #liushanshan
    #print fluid.default_main_program() # print params
    #for param in fluid.default_main_program().global_block().all_parameters(): #print name and shape
    #    print "param name: %s; param shape: %s" % (param.name, param.shape)
    # data reader
    train_reader = data_reader.train(args.batch_size)
    test_reader = data_reader.test()
    # prepare environment
    place = fluid.CPUPlace()
    if args.device >= 0:
        place = fluid.CUDAPlace(args.device)
    exe = fluid.Executor(place)
    exe.run(fluid.default_startup_program())
    train_exe = exe
    error_evaluator.reset(exe)
    if args.parallel:
      train_exe = fluid.ParallelExecutor(
                 use_cuda=True, loss_name=sum_cost.name)
    fetch_vars = [sum_cost] + error_evaluator.metrics

    #inference_program = fluid.io.get_inference_program(error_evaluator)
    inference_program = fluid.default_main_program().clone()

    for pass_id in range(args.pass_num):
        #error_evaluator.reset(exe)
        batch_id = 1
        total_loss = 0.0
        total_seq_error = 0.0
        # train a pass
        save_id = 0
        for data in train_reader():
            var_names = [var.name for var in fetch_vars] 
            if args.parallel:
                 results = train_exe.run(var_names,
                            feed=get_feeder_data(data, place))
                 results = [np.array(result).sum() for result in results]
            else:
                 results = exe.run(
                     fluid.default_main_program(),
                     feed=get_feeder_data(data, place),
                     fetch_list=fetch_vars)
                 results = [result[0] for result in results]
            total_loss += results[0]
            total_seq_error += results[2]
            if batch_id % 10 == 1:
                print '.',
                sys.stdout.flush()
            if batch_id % args.log_period == 1:
                print "\nPass[%d]-batch[%d]; Avg Warp-CTC loss: %s; Avg seq error: %s." % (
                    pass_id, batch_id, total_loss / (batch_id * args.batch_size), total_seq_error / (batch_id * args.batch_size))
                sys.stdout.flush()
            #liushanshan
            if batch_id % args.save_model_period == 0:
                with model_average.apply(exe):
                    error_evaluator.reset(exe)
                    for data in test_reader():
                        exe.run(inference_program, feed=get_feeder_data(data, place))
                    _, test_seq_error = error_evaluator.eval(exe)
                if model_average != None:
                    model_average.restore(exe)
                save_id += 1
                print "\nEnd pass[%d]-batch[%d]; save pass-%05d-%03d:Test seq error: %s.\n" % (
                pass_id, batch_id, pass_id+1, save_id, str(test_seq_error[0]))
                model_path = "./results/" + "pass-%05d-%03d"%(pass_id+1,save_id)
                os.mkdir(model_path)
                fluid.io.save_params(exe, model_path)
                
            batch_id += 1

def main():
    args = parser.parse_args()
    print_arguments(args)
    train(args, data_reader=ctc_reader)

if __name__ == "__main__":
    main()

**Comment by panyx0718** [comment]
@wanghaoshuang can you take a look and see which part of fluid this issue belongs to? So I can assign to the correct people

**Comment by wanghaoshuang** [comment]
@liushanshan07  麻烦你去掉warpctc layer试一下？另外check下你环境下的libcublas.so的版本？

**Comment by shanyi15** [comment]
您好，此issue在近一个月内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持! 
 Hello, this issue has not been updated in the past month. We will close it today for the sake of other user‘s experience. If you still need to follow up on this question after closing, please feel free to reopen it. In that case, we will get back to you within 24 hours. We apologize for the inconvenience caused by the closure and thank you so much for your support of PaddlePaddle Group!
